stdlib/Distributed/src/remotecall.jl

```diff
@@ -247,22 +247,32 @@ function del_clients(pairs::Vector)
     end
 end
 
-const any_gc_flag = Condition()
+const any_gc_flag = Threads.Condition()
 function start_gc_msgs_task()
-    @async while true
-        wait(any_gc_flag)
-        flush_gc_msgs()
+    @async begin
+        lock(any_gc_flag) do
+            wait(any_gc_flag)
+            flush_gc_msgs()
+        end
     end
```

Comment:
Does it suffice that this no longer runs in a `while true` block?

Reply:
ah that's a bug :)